lib/smack/SmackRep.cpp

```diff
@@ -623,10 +743,34 @@ string SmackRep::getPrelude() {
   s << endl;
   s << "// Memory region declarations";
   s << ": " << memoryRegions.size() << endl;
-  for (unsigned i=0; i<memoryRegions.size(); ++i)
-    s << "var " << memReg(i) << ": " << memType(i) << ";" << endl;
+  for (unsigned i=0; i<memoryRegions.size(); ++i) {
+    if (SmackOptions::BitVectors) {
+      if (SmackOptions::InferFieldOverlap){
```

Comment:
These two loops should be combined into one by starting at `j = 0` and continuing while j is smaller than `SmackOptions::InferFieldOverlap ? 4 : 1`.